### DISCUSSION TITLE
Help wanted for route designing

### DISCUSSION THREAD

**Discussion by anonymouscatcher** [vercel/next.js#26815] [Help]
Hello Everyone,

I'm trying to create a layout as Slack with next js, As you know Slack has a right side bar that based on your action it will show a different component, like Viewing profile or Information about a channel. So I'd like to create similar behavior in my app, without next.js router it's easier to achieve it but with next.js router it requires to create more files that need to repeat some of my code.  So I'd like to share my solution maybe there's an easier way. 

This is how pages directory looks with dynamic sidebar: 
```
> pages
  * index.js
  * [sidebarId] 
     * index.js
  * browse 
    * index.js
    * [sidebarId].js
  * about 
    * index.js
    * [sidebarId].js
``` 
Basically I need to repeat each page because of the one single query parameter. for example `index.js` and `[sidebarId]/index.js` is same! and this bothers me little bit.

Note: I have base layout shared by all the pages and inside the layout I can access to `sidebarId` then I can render an specific component.
Let me know if you have better idea.
Thanks in advance

**Comment by LarsEjaas** [comment]
Not to "highjack" your thread, but I am in a similar situation. 

In my case I have this structure:

```
> pages
   * [...slug].tsx
   > [locale]
       > blog
          * [...slug].tsx
```
This works fine when I have a locale, but when I am on the default language my blog post will fall back to pages/[slug...].tsx

It's a shame you can't do optional route parameters - a.k.a a folder like this:   [[locale]]

I am also struggling to find a solution for this!?

**Reply by anonymouscatcher** [reply]
Me too, still struggling with this, I also have another issue imagine you have slug parameters in your route and you want to change the route when you click on a link, you want to only change the route before slug parameters, How will you do that?! 
This is really handy with react router, you can use matchers to detect the params, not sure what’s the best way to do this in next, I read the docs and examples couple of times.

**Reply by anonymouscatcher** [reply]
Won’t solve my issue, but you can have optional catch all routes based on documents 
https://nextjs.org/docs/routing/dynamic-routes#optional-catch-all-routes

**Reply by LarsEjaas** [reply]
Thanks alirezavlz I do that already, but for me that is also part of the problem. Because some of my blogposts from my CMS is wrongly identified as pages when no locally is available (when I am on the default locale).

**Reply by LarsEjaas** [reply]
For now I just have to go with identical files in some of the page folders. That's not ideal at all, as I have to remember to apply any changes to both files.
